src/cppcache/src/TcrConnection.cpp

```diff
@@ -703,15 +717,21 @@ inline ConnErrType TcrConnection::sendData(uint32_t& timeSpent,
            is not public yet*/
         notPublicApiWithTimeout == TcrMessage::EXECUTE_FUNCTION ||
         notPublicApiWithTimeout == TcrMessage::EXECUTE_REGION_FUNCTION ||
-        notPublicApiWithTimeout ==
-            TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP) {
-      // then app has set timeout in millis, change it to microSeconds
-      sendTimeoutSec = sendTimeoutSec * 1000;
-      isPublicApiTimeout = true;
-      LOGDEBUG("sendData2 %d ", sendTimeoutSec);
-    } else {
-      sendTimeoutSec = sendTimeoutSec * 1000;
-    }
+    	notPublicApiWithTimeout == TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP ||
+		// GT GEODE-2891
+		notPublicApiWithTimeout == TcrMessage::HANDSHAKE)
+	{
+	    // then app has set timeout in millis, change it to microSeconds
+	    sendTimeoutSec = sendTimeoutSec * 1000;
+	    isPublicApiTimeout = true;
+	    LOGDEBUG("sendData2 %d ", sendTimeoutSec);
+	} 
```

Comment:
Formatting does not conform to  [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).